plugin/src/main/groovy/com/btkelly/gnag/utils/ReportWriter.java

```diff
@@ -42,29 +41,27 @@
     private static final String HTML_REPORT_SUFFIX =
             "</article></html>";
 
-    public static boolean writeReportToDirectory(
+    public static boolean writeLocalReportFiles(
             @NotNull final Set<Violation> violations,
             @NotNull final File directory) {
+        
+        if (violations.isEmpty()) {
+            throw new IllegalStateException("This method should only be called when violations were detected.");
```

Comment:
I'm not sure I like trowing exceptions in these cases. We risk killing the whole Gnag process when it might be a bit more error tolerant to simply log to the console that this occurred and return. Basically the same as the end of the method does if it fails to write to the file.